lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -88,10 +88,12 @@ private function _constructPdoDsn(array $params)
 
         if (isset($params['dbname'])) {
             $dsn .= 'dbname=' . $params['dbname'] . ' ';
+        } elseif (isset($params['default_dbname'])) {
+            $dsn .= 'dbname=' . $params['default_dbname'] . ' ';
         } else {
```

Comment:
Strongly disagree. As already said this is basically what it's all about. Defining `default_dbname` should be the last resort option in case the database server setup doesn't have the `postgres` database. When going back to the old behaviour, we move the action to be taken to the user which is not good IMO. DBAL should try everything possible first before the user is asked to save the world.